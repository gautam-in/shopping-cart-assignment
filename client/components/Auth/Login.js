import styles from '../../styles/login.module.scss';
import TextField from '@material-ui/core/TextField';
import { useRouter } from 'next/router'
import useForm from '../../util/useForm';

const Login = () => {

    const router = useRouter();
    const { inputs, handleChange, resetForm } = useForm({
        email: '',
        password: ''
    });

    return (
        <div className={styles.loginContainer}>
            <div className={styles.login}>
                <p className={styles.loginTitle}>Login</p>
                <p>Get access to your Orders, Wishlist and Recommendations</p>
            </div>
            <div className={styles.loginForm}>
                <form onSubmit={(e) => {
                    e.preventDefault();
                    console.log(inputs);
                    resetForm();
                    router.push('/')
                }} method="POST">
                    <TextField
                        type="email"
                        required
                        label="Email"
                        id="Email"
                        size="small"
                        name="email"
                        value={inputs.email}
                        onChange={handleChange}
                    />
                    <TextField
                        type="password"
                        required
                        label="Password"
                        size="small"
                        margin="normal"
                        name="password"
                        value={inputs.password}
                        onChange={handleChange}
                    />

                    <button>Login</button>
                </form>
            </div>
        </div>
    )
}

export default Login
