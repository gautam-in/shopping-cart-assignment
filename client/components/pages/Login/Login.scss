@import "/client/styles/variables";
@import "/client/styles/mq-mixin";

.login {
  width: 80%;
  margin: 30px auto;
  padding-top: 30px;

  flex-direction: column;
  /* margin-top: 60px; */
  h1 {
    margin-bottom: 20px;
    font-size: 30px;
  }
  h2 {
    font-size: $primary-font-size;
    color: $font-secondary-color;
    font-weight: 500;
  }
  > div,
  > form {
    margin-bottom: 20px;
    width: 100%;
  }
  > form {
    max-width: 300px;
  }
  @include mq(tab) {
    flex-direction: row;
  }
}
