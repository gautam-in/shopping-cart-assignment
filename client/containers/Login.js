import classes from '../styles/login.module.scss';

const LoginPage = () => {
  return (
    <>
      <section className={classes.container}>
        <section className={classes.loginIntro}>
          <h1>Login</h1>
          <p>Get access to your Orders, Wishlist and Recommendation</p>
        </section>

        <form className={classes.loginForm} action='/' method='post'>
          <label className={classes.inputLabel}>
            <input type='email' required className={classes.input} />
            <span className={classes.placeholder}>Email</span>
          </label>

          <label className={classes.inputLabel}>
            <input type='password' required className={classes.input} />
            <span className={classes.placeholder}>Password</span>
          </label>

          <button type='submit' className={classes.loginButton}>
            Login
          </button>
        </form>
      </section>
    </>
  );
};

export default LoginPage;
