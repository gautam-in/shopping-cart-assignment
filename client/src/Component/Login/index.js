import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import AuthContext from "../../Auth.jsx";
import "./index.scss";

const SignIn = () => {
  const { toggleUserAuthentication } = useContext(AuthContext);
  const history = useHistory();
  
  const initialState = {
    email: "",
    password: "",
  };
  const [userDetails, setUserDetails] = useState(initialState);
  const [validation, setValidation] = useState("");

  const onFormSubmit = (e) => {
    e.preventDefault();
    var regex = "^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$";
    const email = localStorage.getItem("email");
    const pwd = localStorage.getItem("password");
    if (!email && !pwd) {
      setValidation("please register first");
    } else if (!pwd) {
      setValidation("please enter the password");
    } else if (!email) {
      setValidation("please enter the email");
    } else {
      if (
        userDetails.email.match(regex) &&
        userDetails.email.length > 1 &&
        userDetails.password.length > 6 &&
        userDetails.email === email &&
        userDetails.password === pwd
      ) {
        toggleUserAuthentication();
        localStorage.setItem("status", "logged-in");
        history.push("/");
      } else {
        setValidation("Invalid credentials");
      }
    }
  };

  return (
    <div className="sign-in">
      <aside>
        <h1> Login </h1>{" "}
        <p> Get access to your Orders, Wishlist and Recommendations </p>{" "}
      </aside>{" "}
      <section className="register-form">
        <form onSubmit={onFormSubmit}>
          <div class="form__group field">
            <input
              type="Email"
              className="form__field"
              placeholder="email"
              name="email"
              id="email"
              required
              value={userDetails.email}
              onChange={(event) =>
                setUserDetails({
                  ...userDetails,
                  email: event.target.value,
                })
              }
            />{" "}
            <label className="form__label"> Email </label>{" "}
          </div>{" "}
          <div class="form__group field">
            <input
              type="password"
              className="form__field"
              placeholder="password"
              name="password"
              id="password"
              required
              onChange={(event) =>
                setUserDetails({
                  ...userDetails,
                  password: event.target.value,
                })
              }
            />{" "}
            <label className="form__label"> Password </label>{" "}
          </div>{" "}
          <p
            style={{
              color: "red",
              textAlign: "center",
            }}
          >
            {" "}
            {validation}{" "}
          </p>{" "}
          <button className="SignIn-btn"> Login </button>{" "}
        </form>{" "}
      </section>{" "}
    </div>
  );
};

export default SignIn;
