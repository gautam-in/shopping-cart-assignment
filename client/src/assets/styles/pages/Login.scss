@import "../forms.scss";

.page-login {
  .container {
    max-width: 1000px;
    margin: 0 auto;
    padding: 1rem;
    display: grid;
    gap: 16px;
  }

  h1 {
    font-weight: bold;
  }

  .sub-heading {
    font-size: 0.875em;
    margin-top: 1rem;
  }

  .form-grid {
    margin: 1.25rem 0;
    display: grid;
    gap: 24px;
  }
}

@media (min-width: 576px) {
  .page-login {
    .container {
      padding: 2rem 1rem;
      grid-template-columns: repeat(2, 1fr);
    }

    h1 {
      margin-top: 2.5rem;
    }

    .sub-heading {
      margin-top: 1.25rem;
    }
  }
}

@media (min-width: 992px) {
  .page-login {
    .container {
      padding: 3.125rem 1rem;
    }
  }
}
