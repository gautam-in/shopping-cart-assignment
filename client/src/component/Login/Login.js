import {
	SignInContainer,
	GridConatiner,
	GridItem,
	ItemContainer,
	FormTextField,
	RegiterTitle,
	RegiterContent,
	SignUpButton,
	ErrorMessage,
} from '../Register/StyledSignin';
import { authActions } from '../../reducers/UserReducer';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';

const Login = () => {
	const auth = useSelector((state) => state.auth);
	const dispatch = useDispatch();

	useEffect(() => {
		if (auth.status !== null && auth.error !== null)
			dispatch(authActions.resetError());
	}, [dispatch]);

	const [fields, setFields] = useState({
		email: '',
		emailError: null,
		password: '',
		passwordError: null,
	});
	const emailRegex =
		/^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i;
	const passwordRegex = /(?=.*\d)(?=.*[a-z])(?=.[A-Z]).{6,}/;

	const handleChange = (event) => {
		const { name, value } = event.target;
		setFields((state) => ({ ...state, [name]: value }));
	};

	const handleSubmit = (event) => {
		event.preventDefault();

		if (fields.email === '') {
			setFields((state) => ({ ...state, emailError: 'Email is required' }));
		} else if (!fields.email.match(emailRegex)) {
			setFields((state) => ({
				...state,
				emailError: 'Email should be syntax of xxx@x.xxx',
			}));
		}

		if (fields.password === '') {
			setFields((state) => ({
				...state,
				passwordError: 'Password is required',
			}));
		} else if (!fields.password.match(passwordRegex)) {
			setFields((state) => ({
				...state,
				passwordError:
					'Password should contain min 6 characters - numbers,alphabets and cannot have spaces.',
			}));
		}

		if (
			fields.email.match(emailRegex) &&
			fields.password.match(passwordRegex)
		) {
			const credentials = { email: fields.email, password: fields.password };
			setFields((state) => ({
				...state,
				emailError: null,
				passwordError: null,
			}));
			dispatch(authActions.loginUser(credentials));
			console.log(credentials, auth);
		}
	};

	return (
		<>
			<SignInContainer component="form" onSubmit={handleSubmit}>
				<GridConatiner container spacing={1}>
					<GridItem item xs={12} sm={5} md={6}>
						<ItemContainer>
							<RegiterTitle>Login</RegiterTitle>
							<RegiterContent>
								Get Access to your Orders, Wishlist and Recommendations
							</RegiterContent>
							{auth?.status === 'failure' && (
								<ErrorMessage>{auth.error}</ErrorMessage>
							)}
						</ItemContainer>
					</GridItem>
					<GridItem item xs={12} sm={5} md={5} sx={{ mt: '10px' }}>
						<FormTextField
							name="email"
							label="Email"
							type="text"
							variant="standard"
							required
							onChange={handleChange}
							value={fields.email}
							fullWidth
						/>
						<ErrorMessage>{fields.emailError}</ErrorMessage>
						<FormTextField
							name="password"
							label="Password"
							variant="standard"
							type="password"
							required={true}
							onChange={handleChange}
							value={fields.password}
							fullWidth
						/>
						<ErrorMessage>{fields.passwordError}</ErrorMessage>

						<SignUpButton variant="contained" fullWidth type="submit">
							Login
						</SignUpButton>
					</GridItem>
				</GridConatiner>
			</SignInContainer>
		</>
	);
};

export default Login;
