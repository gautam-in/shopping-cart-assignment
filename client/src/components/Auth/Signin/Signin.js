import { useHistory } from "react-router-dom";
import "./signin.scss";
import Input from "../../UI/Input/Input";

const Signin = () => {
  const history = useHistory();

  const submitHandler = (e) => {
    e.preventDefault();
    history.push("/");
  };
  
  return (
    <div className="flex--row">
      <div className="pageDetails">
        <h1 className="pageDetails__heading">Login</h1>
        <p className="pageDetails__desc">Get access to your Orders, Wishlist and Recommendations</p>
      </div>
      <div className="auth-form">
        <form onSubmit={submitHandler} method="POST" action="/">
          <div className="rowInput flex--column">
            <Input
              type="email"
              placeholder="Email"
              className="rowInput__input-text"
              id="email"
              required
              name="email"
            />
            <label htmlFor="email" className="rowInput__labelHelper">
              Email
            </label>
          </div>
          <div className="rowInput flex--column">
            <Input
              type="password"
              name="password"
              id="password"
              className="rowInput__input-text"
              placeholder="Password"
              required
            />
            <label htmlFor="password" className="rowInput__labelHelper">
              Password
            </label>
          </div>
          <div className="submit-btn"> 
          <input className="submit" type="submit" value="Login" />
          </div>
        </form>
      </div>
    </div>
  );
};

export default Signin;