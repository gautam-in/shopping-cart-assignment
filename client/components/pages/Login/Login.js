import React from "react";
import { useHistory } from "react-router";

import useForm from "../../../customHooks/useForm";

import Button from "../../atoms/Button/Button";
import FormField from "../../atoms/FormField/FormField";
import {
  loginButton,
  emailRegex,
  loginEmailLabel,
  loginPasswordLabel,
  emailRegexError,
} from "../../../constant";

import "./Login.scss";

const initialState = { email: "", password: "" };

function Login() {
  const history = useHistory();

  let rules = {
    email: ["required", emailRegex, emailRegexError],
    password: ["required"],
  };

  const { inputs, errors, handleChange } = useForm(initialState, rules);

  const handleLogin = (e) => {
    e.preventDefault();
    console.log("Login called", inputs);
    if (Object.keys(errors).length === 0) {
      history.push("/");
    }
  };

  return (
    <div className="flexed login" data-test="component-login">
      <div>
        <h1>Login </h1>
        <h2>Get access to your Orders, Wishlist and Recommendations</h2>
      </div>
      <form onSubmit={handleLogin} method="POST">
        <FormField
          label={loginEmailLabel}
          type="text"
          name="email"
          id="email"
          value={inputs.email}
          errors={errors.email}
          onChange={handleChange}
          required
        />
        <FormField
          label={loginPasswordLabel}
          type="password"
          name="password"
          id="password"
          value={inputs.password}
          errors={errors.password}
          onChange={handleChange}
          required
        />
        <Button
        //disabled={Object.keys(errors).length !== 0}
        >
          {loginButton}
        </Button>
      </form>
    </div>
  );
}

export default Login;
