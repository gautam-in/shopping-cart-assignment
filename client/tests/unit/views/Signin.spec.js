import { mount, shallowMount } from "@vue/test-utils";
import Signin from "@/views/Signin.vue";
import Login from "@/components/Login.vue";
import Vue from "vue";
import VueRouter from "vue-router";
import Vuelidate from "vuelidate";
Vue.use(Vuelidate);

describe("Signin", () => {
  it("snapshot", () => {
    const wrapper = mount(Signin, {
      stubs: {
        Login,
      },
    });
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.contains("div")).toBe(true);
  });
});
