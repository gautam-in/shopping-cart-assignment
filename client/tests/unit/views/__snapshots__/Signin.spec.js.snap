// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin snapshot 1`] = `
<div class="signin">
  <section id="loginContainer">
    <div class="container">
      <div class="row">
        <div class="loginInfo col-lg-6 col-xl-6 col-md-6 col-sm-12 col-12">
          <h2>Login</h2>
          <p>Get access to your Orders, Wishlish and Recommendations</p>
        </div>
        <div class="col-lg-6 col-xl-6 col-md-6 col-sm-12 col-12">
          <div class="mb-3 row"><label for="staticEmail" class="col-sm-2 col-form-label">Email</label>
            <div class="col-lg-4 col-xl-7 col-md-4 col-sm-10"><input type="text" id="staticEmail" class="form-control">
              <!---->
            </div>
          </div>
          <div class="mb-3 row"><label for="inputPassword" class="col-sm-2 col-form-label">Password</label>
            <div class="col-lg-4 col-xl-7 col-md-4 col-sm-10"><input type="password" id="inputPassword" class="form-control">
              <!---->
            </div>
          </div>
          <div class="mb-3 row">
            <div class="btnContainer col-lg-10 col-xl-10 col-md-4 col-sm-12 col-12"><button class="btnStyle">
                Login
              </button></div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
